lib/pos_mempool_persister.go

```diff
@@ -0,0 +1,205 @@
+package lib
+
+import (
+	"bytes"
+	"github.com/deso-protocol/core/storage"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"sync"
+	"time"
+)
+
+const (
+	DbMempoolContextId = "transactions"
+	eventQueueSize     = 10000
+)
+
+type MempoolEventType int
+
+const (
+	MempoolEventAdd MempoolEventType = iota
+	MempoolEventRemove
+	MempoolEventExit
+)
+
+type MempoolPersisterStatus int
+
+const (
+	MempoolPersisterStatusRunning MempoolPersisterStatus = iota
+	MempoolPersisterStatusNotRunning
+)
+
+// MempoolPersister is responsible for persisting transactions in the mempool to the database. Whenever a transaction is
+// added or removed from the mempool, the MempoolPersister is notified via the OnMempoolEvent callback. The Persister
+// will then add the event to a queue. Periodically, the transaction queue is flushed to the database and all the cached
+// transactions are persisted.
+type MempoolPersister struct {
+	sync.Mutex
+	stopGroup  sync.WaitGroup
+	startGroup sync.WaitGroup
+
+	frequencyMilliseconds int
```

Comment:
Always nice to clarify naming
```suggestion
	batchPersistFrequencyMilliseconds int
```